From ca0522b59ebb4e494116b241e91269f96df27c9d Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Mon, 20 May 2019 06:25:31 -0600
Subject: [PATCH 4/6] Snap: Save SDCard photos to legacy path

Changes to fw/b make this possible again.

This is a squash of:

* Revert "Fix FC of external SD Card path null"

  This reverts commit 6efe10b068793d6180c851b7443bea2434df47aa.

* Revert "Fix issue of can't save image in sdcard on P"

  This reverts commit 9b4755c0ce090a14dfb7599a238b044dcdfd5ee4.

Change-Id: If9721dad3cf63c2d4e155827676f5e78eeb821d9
---
 src/com/android/camera/SDCard.java | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/src/com/android/camera/SDCard.java b/src/com/android/camera/SDCard.java
index be839e2b4..b88e32245 100644
--- a/src/com/android/camera/SDCard.java
+++ b/src/com/android/camera/SDCard.java
@@ -32,20 +32,16 @@ import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
 import android.content.IntentFilter;
-import android.os.Build;
 import android.os.Environment;
 import android.os.storage.StorageVolume;
 import android.os.storage.StorageManager;
 import android.util.Log;
 
-import java.io.File;
-
 public class SDCard {
     private static final String TAG = "SDCard";
 
     private static final int VOLUME_SDCARD_INDEX = 1;
 
-    private Context mContext;
     private StorageManager mStorageManager = null;
     private StorageVolume mVolume = null;
     private String mPath = null;
@@ -66,18 +62,7 @@ public class SDCard {
             return null;
         }
         if (mPath == null) {
-            File[] dirs = mContext.getExternalFilesDirs(null);
-            if (dirs != null) {
-                String dir;
-                for (int i=0; i<dirs.length; i++) {
-                    if (dirs[i] == null) continue;
-                    dir = dirs[i].getAbsolutePath();
-                    if (dir.startsWith(mVolume.getPath())) {
-                        mPath = dir;
-                        break;
-                    }
-                }
-            }
+            mPath = mVolume.getPath() + "/DCIM/Camera";
         }
         return mPath;
     }
@@ -108,7 +93,6 @@ public class SDCard {
 
     private SDCard(Context context) {
         try {
-            mContext = context;
             mStorageManager = (StorageManager) context.getSystemService(Context.STORAGE_SERVICE);
             initVolume();
             registerMediaBroadcastreceiver(context);
-- 
2.17.1

